rippled v1.1.0 Changes [Waiting for official release]

mDuo13 opened the pull request:
Combines the following PRs with changes relating to `rippled` 1.1.0:

- #438 (DepositPreauth reference changes)
- #439 (Ubuntu build tutorial updates)
- #443 (SQLite transaction db page size issue and fix)
- #446 (`--unittest-ipv6` commandline option)
- https://github.com/mDuo13/ripple-dev-portal/pull/46 (signing command permission changes)

Also adds a couple minor changes:

- Update Ubuntu instructions to more clearly specify that it requires Boost 1.67.0 exactly (since Boost 1.68.0 is not compatible with rippled 1.1.0)
- Use `master` instead of `develop` for rippled source code links. Update a few line numbers accordingly. (The line numbers for DepositPreauth objects are speculatively set to match what's expected to be merged to master based on the release candidate.)
- Update the `deposit_authorized` method's error codes to match https://github.com/ripple/rippled/pull/2642

This PR needs minimal modifications after the `rippled` 1.1.0 release:

- [ ] Update the build instructions with the version-setting commit for v1.1.0
- [ ] Re-run the link checker to confirm that all the v1.1.0-related links work

mDuo13 committed:
DepositPreauth reference changes (draft, missing examples)

mDuo13 committed:
Preauth-related API method docs (draft)

mDuo13 committed:
Add examples to DepositPreauth reference docs & clarify some details

mDuo13 committed:
Update Ubuntu buildtutorial for rippled v1.1.0

jacksteroo committed:
Update ubuntu build steps

- consolidated initial install dependencies
- BOOST_ROOT dir will be incorrectly set if your username is not ubuntu
- (Revoved by @mduo13: automatically setting number of processors to use to compile Boost libraries to half of what users would have)
- git log to now show v1.0.1
- numbering changes

mDuo13 committed:
Edits to preauth reference

Commit:
Add tutorial for increasing transaction db page size.

mDuo13 committed:
Add --unittest-ipv6 option to CLI ref

mDuo13 committed:
Add sqlite page size description to dactyl config

mDuo13 committed:
Rewrite SQLite transaction db page size issue tutorial

mDuo13 committed:
Fix styling of plain-text code blocks

mDuo13 committed:
Merge branch 'preauth_reference' into rippled_v110

mDuo13 committed:
Merge branch 'doc1673' into rippled_v110

mDuo13 committed:
Merge branch 'mtrippled/page' into rippled_v110

mDuo13 committed:
Merge branch 'doc1650' into rippled_v110

mDuo13 committed:
Build instructions: modify Boost 1.67 note

Boost 1.68.0 is not compatible with rippled 1.1.0 due to some changes to
Boost.Beast. Versions of Boost older than 1.67.0 are not compatible with rippled
1.1.0 because they do not include Boost.Beast. Only Boost 1.67.0 is
compatible, so I updated the build instructions to be clearer about
this.

mDuo13 committed:
Update rippled source links to use master branch

mDuo13 committed:
Update deposit_authorized method error messages

Addresses changes implemented in https://github.com/ripple/rippled/pull/2642

- Missing fields result in `invalidParams`
- Accounts not found in the ledger result in `{src/dst}ActNotFound`

mDuo13 commented:
@jbheron If you could, please peer review just [`fix-sqlite-tx-db-page-size-issue.md`](https://github.com/ripple/ripple-dev-portal/pull/448/commits/8028d18b40d7c62493cffb45a6d2dab59d821edd)

ryangyoung commented on content/tutorials/manage-the-rippled-server/fix-sqlite-tx-db-page-size-issue.md:
```diff
@@ -0,0 +1,173 @@
+
+`rippled` servers with full transaction history (or a very large amount of transaction history) and a database that was initially created with a `rippled` version earlier than 0.40.0 (released January 2017) may encounter a problem with their SQLite database page size that stops the server from operating properly. Servers that store only recent transaction history (the default configuration) and s ...
+
+This document describes steps to detect and correct this problem if it arises.
```
arises --> occurs?

ryangyoung commented on content/tutorials/manage-the-rippled-server/fix-sqlite-tx-db-page-size-issue.md:
```diff
@@ -0,0 +1,173 @@
+
+## Background
+
+`rippled` servers store a copy of their transaction history in a SQLite database. Before version 0.40.0, `rippled` configured this database to have a capacity of roughly 2TB. For most uses, this is plenty. However, full transaction history back to ledger 32570 (the oldest ledger version available in the production XRP Ledger history) is likely to exceed this exceed the SQLite database capacity. ` ...
```
> ledger 32570 (the oldest ledger version available in the production XRP Ledger history

Might be a good place to link to the explanation of this to your footnote in the consensus intro.

mDuo13 replied:
I'm going to hold off on that for now, because:

- That content isn't in this branch yet because it's from a PR started after this one. I'd have to rebase onto master.
- I don't think the link adds much info beyond what's already mentioned in the parenthesis, and it might distract from the task at hand.

ryangyoung reviewed (commented):
The SQLite doc looks great! 👍

mDuo13 committed:
SQLite page size fix: arises→occurs

mDuo13 committed:
Update signing command documentation

Per https://github.com/ripple/rippled/pull/2657, the signing commands
should not be used as public methods and this usage is disabled in
rippled v1.1.0 and later.

mDuo13 committed:
Merge pull request #46 from mDuo13/signing_changes

Update signing command documentation

mDuo13 commented:
Per https://github.com/ripple/rippled/pull/2657, the signing commands should not be used as public methods and this usage is disabled in rippled v1.1.0 and later. The latest commit addresses this:

- Adds a short tutorial for how to turn public signing back on
- Adds a note about this to the signing methods
- Fixes #415 by matching the descriptions of the signing fields between `sign` and `sign_for`.

I left the signing methods in the "public" category for the following reasons:

- That's where they were before, so it's friendlier to users not to move things if we don't have to. (Technically, the file names won't change so it wouldn't likely break any bookmarks to move them, but still.)
- They _can_ be used as public methods. (If anything, they belong in both places, but that would be weird, too.)
- Conceptually, it makes sense for the signing and submitting methods to be grouped together. You can get a sense of the flow for a transaction just by looking at the contents of that category.

mDuo13 committed:
SQLite doc: fix broken 'update rippled' link

mDuo13 committed:
Update to v1.1.0 commit message
